### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by tastybug**
Improve README:
* introduce all event types, give pointers to relevant code sections
* give instructions on how to set up shop as a developer

Improve swagger: Swagger can't deal with generic return types, which is why I removed `EventWrapper` and replaced it with `DealerEventEnvelope`, `ListingEventEnvelope`, `EnrichedListingEventEnvelope` and `ListingUrlEventEnvelope`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,74 @@
...
 
-### How to experiment with this
+### Running locally
 
 The service is stateless and can be started locally by running `./gradlew clean bootRun`, after which you can open <http://localhost:8080/swagger-ui.html>.
 
-This will open Swagger UI, which is a bare bone REST client that you can use to publish events into the service and see how it reacts.
+This will open Swagger UI, which is a simple REST client that you can use to publish events into the service and see how it reacts.
 
 At the bottom of the page your can browse through the MoVe API model, complete with examples, validation rules and description of semantics. Make sure to check those out to map incoming data to your internal model.
```
Comment (markus-schnelle) [inline README.md:24]:
```suggestion
At the bottom of the page you can browse through the MoVe API model, complete with examples, validation rules and description of semantics. Make sure to check those out to map incoming data to your internal model.
```

File: src/main/java/org/example/move/webhookreceiver/rest/listing/ListingReceiverController.java [hunk 2]
```diff
@@ -4,30 +4,26 @@
...
 import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestHeader;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 @RequiredArgsConstructor
 @RequestMapping("/")
-@Api(tags = {"Listing Lifecycle Event Receiver"})
+@Api(tags = {"Listing Lifecycle Event Receiver"}, description = "Receive listing updates. Comes in the flavors \"listing\" and \"enriched listing\", which also contains associated dealer data and promotions.")
```
Comment (markus-schnelle) [inline src/main/java/org/example/move/webhookreceiver/rest/listing/ListingReceiverController.java:26]:
```suggestion
@Api(tags = {"Listing Lifecycle Event Receiver"}, description = "Receive listing updates. Comes in the flavors \"listing\" or \"enriched listing\", which also contains associated dealer data and promotions.")
```

File: src/test/java/org/example/move/webhookreceiver/rest/ListingUrlEventIntegrationTest.java [hunk 3]
```diff
@@ -75,10 +75,10 @@ void dealer_update_can_be_posted() throws JsonProcessingException {
             Void.class);
     }
 
-    private EventWrapper<ListingUrl> aListingUrlEvent() {
+    private ListingUrlEventEnvelope aListingUrlEvent() {
 
         try {
-            TypeReference<EventWrapper<ListingUrl>> eventWrapperTypeReference = new TypeReference<>() {
+            TypeReference<ListingUrlEventEnvelope> eventWrapperTypeReference = new TypeReference<>() {
```
Comment (markus-schnelle) [inline src/test/java/org/example/move/webhookreceiver/rest/ListingUrlEventIntegrationTest.java:81]:
```suggestion
            TypeReference<ListingUrlEventEnvelope> eventEnvelopeReference = new TypeReference<>() {
```

**Review by markus-schnelle — APPROVED** [review]
LGTM, just a few namings. :)

**Comment by tastybug** [comment]
> LGTM, just a few namings. :)

Good feedback, thanks :)

**Pull request merged**